nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RProxyFacetImpl.java

```diff
@@ -72,12 +72,12 @@ protected void doValidate(final Configuration configuration) throws Exception {
 
   @Nullable
   @Override
-  protected Content getCachedContent(final Context context) throws IOException {
+  protected Content getCachedContent(final Context context) {
     AssetKind assetKind = context.getAttributes().require(AssetKind.class);
     TokenMatcher.State matcherState = matcherState(context);
     switch (assetKind) {
-      case PACKAGES:
-        return getAsset(packagesPath(path(matcherState)));
+      case METADATA:
+        return getAsset(extractFullPath(context));
       case ARCHIVE:
         return getAsset(path(path(matcherState), filename(matcherState)));
```

Comment:
the logic related to path looks too complicated. Please create a technical debt task to refactor this. It looks like a good task for an improvement day

Reply:
Agree with you. Here's the ticket https://issues.sonatype.org/browse/NEXUS-21202